### PR TITLE
feat(deslop): consume analyzer slop-fixes + slop-targets

### PR DESCRIPTION
**Pull request by avifenesh**
Companion to [agent-sh/agent-analyzer#27](https://github.com/agent-sh/agent-analyzer/pull/27) + [agent-sh/repo-intel#16](https://github.com/agent-sh/repo-intel/issues/16). Wires the analyzer's new slop queries into the deslop pipeline so the most mechanical work flows through without re-running detection, and the regex/AST scan only looks at files where slop is likely.

## Why

Per the design alignment in this work, deslop's tiers map cleanly to the analyzer outputs:

| Tier | Model | Source |
|---|---|---|
| HIGH (auto-applied with confirmation) | Haiku | analyzer `slop-fixes` |
| MEDIUM (file-level scan) | Sonnet | analyzer `slop-targets` (Sonnet rows) |
| LOW (cross-file judgment) | Opus | analyzer `slop-targets` (Opus rows) |

The analyzer pre-locates the obvious cases (tracked artifacts, orphan exports, empty catches, tautological tests) so deslop doesn't waste detection work on them. For everything else, the analyzer ranks files by slop likelihood with `suspect` labels — deslop only scans those files instead of the full tree.

## What's in

**New module `lib/repo-intel-signals.js`**:

- `getSlopFixes(cwd)` — pinpoint structured fix actions; returns `null` when the analyzer or repo-intel artifact is unavailable
- `getSlopTargets(cwd, {top})` — ranked Sonnet/Opus scan candidates
- `toDeslopFix(analyzerFix)` — shape adapter to deslop's existing fix wire format (`delete-file` / `remove-line` / `replace`), tagged `source: "analyzer-slop-fixes"` for traceability
- `targetsToFileList(targetsResult)` — dedup + sort by score for use as `targetFiles` input to the existing detection pipeline

**Pipeline integration (`scripts/detect.js`)**:

- Pulls both queries before `runPipeline()`
- Passes targets list as `targetFiles` so the regex/AST scan only touches likely-slop files
- Prepends analyzer fixes into `result.fixes` with source tag so consumers see one unified list

**`skills/deslop/SKILL.md`**:

- description trigger list expanded for the new categories
- Phase 0 rewritten around `slop-fixes` + `slop-targets`; old removed `recent-ai` reference cleaned up
- HIGH certainty section now distinguishes analyzer-supplied vs pipeline-detected
- Phase 2 risk weighting uses suspect labels from `slop-targets`

**`agents/deslop-agent.md`**: File Targeting section updated to reflect the two new queries.

## Tests

- 11 new unit tests in `__tests__/repo-intel-signals.test.js`:
  - state-dir resolution (`.claude` / `.opencode` / `.codex`)
  - fix-action shape mapping for all three action types
  - target list dedup + sort
  - graceful null handling when artifact absent
- All 106 existing tests still pass

## Refs

- agent-sh/agent-analyzer#27 — analyzer queries (5 commits, ~5500 lines)
- agent-sh/repo-intel#17 — skill orchestration of the embedder

## Test Plan

- [x] `npm test` passes
- [x] `lib/repo-intel-signals` requires cleanly
- [x] Module gracefully no-ops when no repo-intel
- [ ] CI green
- [ ] End-to-end smoke once analyzer queries land

---

> [!NOTE]
> **Medium Risk**
> Moderate risk because it changes how the analyzer binary and repo-intel artifacts are resolved at runtime (new `lib/agentsys` lookup + removal of vendored `lib/binary`/`lib/repo-map`), which could affect environments where `agentsys` isn’t installed or differs in version.
> 
> **Overview**
> Deslop now consumes analyzer-provided repo-intel signals: it queries `slop-fixes` to inject **pre-located HIGH-certainty fixes** directly into `result.fixes`, and queries `slop-targets` to derive `targetFiles` so the regex/AST scan focuses on likely-slop files.
> 
> Adds `lib/repo-intel-signals` as a small adapter for running these analyzer queries (with graceful `null` fallback), mapping analyzer fix actions into deslop’s fix wire format and extracting a deduped, score-sorted file list.
> 
> Replaces per-plugin vendored analyzer plumbing with a new runtime `lib/agentsys` resolver that locates the canonical `agentsys/lib` install and provides `binary`/`repoIntel`/`repoMap`; collectors and `/deslop` are updated to use it (and now log why repo-intel prefetch was skipped instead of silently ignoring errors). Includes new unit/smoke tests for the signals adapter and the resolver, plus docs updates describing the new Phase 0 behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 0b9c14d9fa95f700661cd0ac27859007bd0ec5a6. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: lib/repo-intel-signals.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        line: analyzerFix.lines && analyzerFix.lines[0],
+        endLine: analyzerFix.lines && analyzerFix.lines[1]
+      });
+    case 'replace-lines':
+      return Object.assign({}, baseFix, {
+        fixType: 'replace',
+        line: analyzerFix.lines && analyzerFix.lines[0],
+        endLine: analyzerFix.lines && analyzerFix.lines[1],
+        replacement: analyzerFix.with
+      });
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply lib/repo-intel-signals.js:142]: Fixed in d86b207 — toDeslopFix now validates that lines is an Array of length >=2 with finite numbers; returns null otherwise (same handling as unknown actions). delete-file also requires a non-empty path. Added 3 test cases for these edge conditions.

File: lib/drift-detect/collectors.js [hunk 2]
```diff
@@ -3,7 +3,7 @@
  * Pure JavaScript data collection - no LLM needed
  *
  * BACKWARD COMPATIBILITY: This module re-exports from lib/collectors.
- * For new code, use: const { collectors } = require('@agentsys/lib');
+ * For new code, use: const { collectors } = require('./lib');
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply lib/drift-detect/collectors.js]: Fixed in d86b207 — corrected to require('..') which resolves to lib/index.js as intended.

File: lib/index.js [hunk 3]
```diff
@@ -25,12 +25,18 @@ const customHandler = require('./sources/custom-handler');
...
-
+// Resolve the canonical agentsys binary at first call. Lazy so
+// modules other than this one don't pay the lookup cost.
+let binaryModule = null;
+function binary() {
+  if (!binaryModule) {
+    binaryModule = require('../agentsys').get().binary;
+  }
+  return binaryModule;
+}
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply lib/index.js]: Fixed in d86b207 — removed the dead binaryModule and binary() declarations entirely. Nothing references them and the wrong require path was not exercised; cleaner to delete than to repair-and-leave-unused.
